packages/core/src/modules/discover-features/DiscoverFeaturesApi.ts

```diff
@@ -1,101 +1,144 @@
-import type { AgentMessageProcessedEvent } from '../../agent/Events'
-import type { ParsedMessageType } from '../../utils/messageType'
+import type { DiscoverFeaturesDisclosureReceivedEvent } from './DiscoverFeaturesEvents'
+import type { DiscloseFeaturesOptions, QueryFeaturesOptions, ServiceMap } from './DiscoverFeaturesModuleOptions'
+import type { Feature } from './models'
+import type { DiscoverFeaturesService } from './services'
 
 import { firstValueFrom, of, ReplaySubject, Subject } from 'rxjs'
 import { catchError, filter, map, takeUntil, timeout } from 'rxjs/operators'
 
 import { AgentContext } from '../../agent'
-import { Dispatcher } from '../../agent/Dispatcher'
 import { EventEmitter } from '../../agent/EventEmitter'
-import { filterContextCorrelationId, AgentEventTypes } from '../../agent/Events'
 import { MessageSender } from '../../agent/MessageSender'
 import { createOutboundMessage } from '../../agent/helpers'
 import { InjectionSymbols } from '../../constants'
+import { AriesFrameworkError } from '../../error'
 import { inject, injectable } from '../../plugins'
-import { canHandleMessageType, parseMessageType } from '../../utils/messageType'
 import { ConnectionService } from '../connections/services'
 
-import { DiscloseMessageHandler, QueryMessageHandler } from './handlers'
-import { DiscloseMessage } from './messages'
-import { DiscoverFeaturesService } from './services'
+import { DiscoverFeaturesEventTypes } from './DiscoverFeaturesEvents'
+import { V1DiscoverFeaturesService, V2DiscoverFeaturesService } from './protocol'
 
+export interface DiscoverFeaturesApi<DFSs extends DiscoverFeaturesService[]> {
+  queryFeatures(options: QueryFeaturesOptions<DFSs>): Promise<Feature[] | void>
+  discloseFeatures(options: DiscloseFeaturesOptions<DFSs>): Promise<void>
+}
 @injectable()
-export class DiscoverFeaturesApi {
+export class DiscoverFeaturesApi<
+  DFSs extends DiscoverFeaturesService[] = [V1DiscoverFeaturesService, V2DiscoverFeaturesService]
+> implements DiscoverFeaturesApi<DFSs>
+{
   private connectionService: ConnectionService
   private messageSender: MessageSender
-  private discoverFeaturesService: DiscoverFeaturesService
   private eventEmitter: EventEmitter
   private stop$: Subject<boolean>
   private agentContext: AgentContext
+  private serviceMap: ServiceMap<DFSs>
 
```

Comment:
In other modules I always expose the config as read only property on the API. This was you can access the config later on (and we may add a way to update certain properties at runtime)